ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -383,4 +428,21 @@ public void appendFrom(ReadableByteChannel channel, long size) throws StoreExcep
       }
     }
   }
+
+  /**
+   * A local LRU cache of recent blobs processed by this store.
+   */
+  private class RecentBlobCache extends LinkedHashMap<String, BlobState> {
+    private final int maxEntries;
+
+    public RecentBlobCache(int initialCapacity, int maxEntries) {
```

Comment:
The intent is to use the insertion ordered LinkedHashMap not the accessed-order LinkedHashMap, right?

Reply:
Changed to access order and added test.